components/script/dom/bluetoothremotegattcharacteristic.rs

```diff
@@ -214,6 +214,50 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
             BluetoothRequest::WriteValue(self.get_instance_id(), value, sender)).unwrap();
         return p;
     }
+
+    #[allow(unrooted_must_root)]
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-startnotifications
+    fn StartNotifications(&self) -> Rc<Promise> {
+        let p = Promise::new(&self.global());
+        let p_cx = p.global().get_cx();
+        if uuid_is_blocklisted(self.uuid.as_ref(), Blocklist::Reads) {
+            p.reject_error(p_cx, Security);
+            return p;
+        }
+        if !(self.Properties().Notify() ||
+             self.Properties().Indicate()) {
+            p.reject_error(p_cx, NotSupported);
+            return p;
+        }
+        if !self.Service().Device().Gatt().Connected() {
+            p.reject_error(p_cx, Network);
+            return p;
+        }
+        let sender = response_async(&p, self);
+        self.get_bluetooth_thread().send(
+            BluetoothRequest::EnableNotification(self.get_instance_id(),
+                                                 true,
+                                                 sender)).unwrap();
+        return p;
+    }
+
+    #[allow(unrooted_must_root)]
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-stopnotifications
+    fn StopNotifications(&self) -> Rc<Promise> {
+        let p = Promise::new(&self.global());
+        let p_cx = p.global().get_cx();
+
+        if !self.Service().Device().Gatt().Connected() {
```

Comment:
You are right about this. We differ from the spec, because we do not support any notification context (yet). The reason I put this here to have at least one "early return". But i can remove it if you prefer.